### ISSUE TITLE
Result reproduction

### ISSUE THREAD

**Issue by simicvm** [junhyukoh/value-prediction-network#1]
First, thanks for the paper and code. I guess this is not the issue with the code that I have, but I would be very grateful to get some clarification. Right now I am trying to reproduce results from the paper with the script `python train.py --config config/collect_deterministic.xml --branch 4,4,4,1,1 --decay 0.8 --alg VPN --log result/vpn5 ${@:1}` since I guess that represent VPN(5) from the paper. 

1. Are the default parameters for `lr = 1e-4` and `decay = 0.95` the ones you got the best results with?  
2. Can you tell me some approximate time it took you to finish 60 epochs (15e6 steps) on the deterministic maze environment? I am running on the Intel Xeon CPU E5-2690 v3 (MS Azure VM), and it goes very slowly. After 20h of training, I am just at step 1e6. Also my reward curve almost flattened just shy of 8.

**Comment by junhyukoh** [comment]
1. The hyperparameters in the script (decay = 0.8 for VPN(5)) should reproduce the result of the paper.
2. It took roughly 24-30 hours to finish 60 epochs (15e6 steps) on the maze environment using 4 Titan X GPUs. Using GPUs is a lot faster than using CPUs especially for VPN architectures. The learning curve on your Tensorboard is showing the performance of epsilon-greedy policy. Epsilon decreases from 1.0 to 0.5 for the first 1e6 steps. This is why you see a dramatic change at 1e6 steps. However, it should be actually slowly improving over time. It would be better to look at `eval.csv` in the log folder, which shows the actual performance (greedy policy without epsilon).

**Comment by simicvm** [comment]
Thanks for the response.

As for the reward curve, I am always plateauing around 8, after approximately 1e6 steps. Tried several combinations of parameters, and cpu/gpu workers, nothing seems to help.

**Comment by junhyukoh** [comment]
Hi @simama,
I just realized that `prediction-step` was not set in the `train_vpn` script.
I fixed it and am running vpn(5) to make sure that everything is correct.
I will let you know when I get the result! Thanks.

**Comment by simicvm** [comment]
Thanks for letting me know. It's set by default to 3, but should be 5 for VPN(5), right?

**Comment by simicvm** [comment]
Hi @junhyukoh,
There might be another mismatch between code implementation and paper. Both collect xml files define default reward/penalty as -0.2, while in paper (4.2 Collect Domain) you state that default penalty is -0.02. Couldn't find any place in code where you override that. Am I missing something?

**Comment by junhyukoh** [comment]
Hi @simama,
I am sorry for my late response.
There was a typo in my paper, the penalty should be -0.2 (The code is correct).
Also, the number of prediction steps should be 5 for VPN(5) as you said.
I updated the code a few days ago and confirmed that VPN(5) results are reproducible.
VPN(5) takes around 20-24 hours to train using 4 Titan X GPUs. 
Please make sure to look at `eval.csv` rather than tensorboard learning curves.

**Comment by junhyukoh** [comment]
Here is my `eval.csv` file for VPN(5). 
```
0, -8.710
1, -3.445
2, 3.743
3, 6.142
4, 7.808
5, 8.269
6, 8.494
7, 8.548
8, 8.779
9, 8.885
10, 8.796
11, 8.915
12, 8.684
13, 8.776
14, 8.907
15, 8.884
16, 8.973
17, 9.060
18, 9.006
19, 8.998
20, 8.860
21, 8.872
22, 8.940
23, 9.144
24, 9.031
25, 9.040
26, 9.137
27, 9.199
28, 9.172
29, 9.203
30, 9.210
31, 9.126
32, 9.210
33, 9.114
34, 9.193
35, 9.165
36, 9.156
37, 9.231
38, 9.262
39, 9.272
40, 9.257
41, 9.256
42, 9.263
43, 9.267
44, 9.278
45, 9.294
46, 9.305
47, 9.316
48, 9.309
49, 9.300
50, 9.300
51, 9.297
52, 9.288
53, 9.289
54, 9.291
55, 9.301
56, 9.303
57, 9.288
58, 9.301
59, 9.291
60, 9.297
```

**Comment by simicvm** [comment]
@junhyukoh Thank you. I get something along those lines too, so it's ok. 

Can you tell me also what are the parameters for Atari env (MsPacman)? I am trying to run it with this, based on the info from the paper:
```
python train.py --config config/collect_deterministic.xml --branch 4,4,4,1,1 --decay 0.8 --gpu 0 --prediction-step 3 --alg VPN --log result/pacman_baseline ${@:1} -ps 1 -w 8 --buf 0 -e MsPacman-v0 -g 0.99 --f-num 16,32 --f-stride 4,2 --f-size 8,4 --dim 128
```
But I encounter an issue. Log says I am making the right environment `Making new env: MsPacman-v0`, but the returned `env` object doesn't have many methods like `meta()`, `tf_visualize()`, etc, which causes crash every time. Should I go into scripts and try to find every one of these method calls that are causing problems, or am I not calling the `train.py` properly? I don't see anything in `parser` arguments that could make any difference with this problem.
